### PR TITLE
feat(i-373): create support for Immutable

### PR DESCRIPTION
**Pull request by gsi-ariam**

### CODE REVIEW

File: __test__/schema-inmutable.spec.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    const CardSchema = new Schema(CardSchemaBase, { strict: false });
+    const Card = model('Card', CardSchema);
+    await startInTest(getDefaultInstance());
+    const result = await Card.create(cardInfo);
+    result.cardNumber = '80';
+    expect(result.cardNumber).toBe('80');
+  });
+
+  test("Test Schema Immutable integration on strict='throw'", async () => {
+    const CardSchema = new Schema(CardSchemaBase, { strict: 'throw' });
```
Comment (gsi-chao) [inline __test__/schema-inmutable.spec.ts:28]: What is strict: "throw" strict is a boolean field, Maybe you want to use strategy: CAST_STRATEGY. THROW

File: src/model/create-model.ts [hunk 2]
```diff
@@ -96,7 +96,7 @@ export const _buildModel = (metadata: ModelMetadata) => {
       const strategy = options.strategy || CAST_STRATEGY.DEFAULT_OR_DROP;
       const strict = options.strict !== undefined ? options.strict : schema.options.strict;
       const skip = options.skip || [modelKey, ID_KEY];
-      const schemaData = cast(data, schema, { strategy, strict, skip });
+      const schemaData = cast(data, schema, { strategy, strict: !!strict, skip });
```
Comment (gsi-chao) [inline src/model/create-model.ts:99]: You have the wrong concept of strict. Review the documentation in this section https://v2.ottomanjs.com/guides/schema.html#cast-method

File: src/model/document.ts [hunk 3]
```diff
@@ -296,8 +333,23 @@ export abstract class Document<T> {
    * ```
    */
   _applyData(data) {
+    const strict = this.$.schema.options.strict;
     for (const key in data) {
       this[key] = data[key];
+      const isImmutable = (this.$.schema.fields[key] as any)?.options?.immutable;
+      if (!this.immutableHasOwnProperty(key) && isImmutable && strict) {
```
Comment (gsi-chao) [inline src/model/document.ts:340]: Check the strict logic here too.

File: src/schema/interfaces/schema.types.ts [hunk 4]
```diff
@@ -44,7 +45,7 @@ export abstract class IOttomanType {
 }
 
 export interface SchemaOptions {
-  strict?: boolean;
+  strict?: boolean | 'throw';
```
Comment (gsi-chao) [inline src/schema/interfaces/schema.types.ts:48]: Review this too.

File: src/schema/schema.ts [hunk 5]
```diff
@@ -88,7 +88,7 @@ export class Schema {
     const preHooks = options?.preHooks;
     const postHooks = options?.postHooks;
     this.options = options;
-    const strict = options?.strict || false;
+    const strict = !!options?.strict || false;
```
Comment (gsi-chao) [inline src/schema/schema.ts:91]: Review this too.

File: src/schema/schema.ts [hunk 6]
```diff
@@ -120,7 +120,7 @@ export class Schema {
   validate(data: unknown, options: { strict?: boolean } = {}) {
     const _options = {
       strategy: CAST_STRATEGY.THROW,
-      strict: options.strict !== undefined ? options.strict : this.options.strict,
+      strict: options.strict !== undefined ? !!options.strict : !!this.options.strict,
```
Comment (gsi-chao) [inline src/schema/schema.ts:123]: Review this too.

**Review by gsi-chao — CHANGES_REQUESTED** [review]

**Review by gsi-chao — APPROVED** [review]
Looks good for me

**Review by gsi-alejandro — APPROVED** [review]

**Review by httpJunkie — APPROVED** [review]

**Pull request merged**